atlasdb-api/src/main/java/com/palantir/atlasdb/spi/TransactionKeyValueServiceManagerFactory.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.spi;
+
+import com.palantir.atlasdb.coordination.CoordinationService;
+import com.palantir.atlasdb.transaction.api.TransactionKeyValueServiceManager;
+import com.palantir.refreshable.Refreshable;
+
+public interface TransactionKeyValueServiceManagerFactory<T> {
+
+    String getType();
+
+    Class<T> coordinationClass();
+
+    TransactionKeyValueServiceManager createManager(
```

Comment:
Might want a bit of docs explaining what the non-obvious params might be used for (mainly CoordinationService)

Reply:
Hmm, didn't see this - did you forget to push?